Add the remaining placeheld policies and reorganize to implementation specific directory

komish opened the pull request:
This PR migrates the remaining policies implemented on original PoCs, but with placeholder actions for the moment as some of them may not be in immediate scope. 

In addition, this PR reorganizes the location of the underlying implements to a subdirectory called `podmanexec`, separating the internal policy interface and struct definitions from the implementations themselves. This should allow us to just "create a new directory and start writing implementations" in a situation where we want to change the underlying implementation details (which we're currently researching by way of libpod, for example).

Signed-off-by: Jose R. Gonzalez <josegonzalez89@gmail.com>

komish committed:
add the remaining placeheld policies and reorganize to implementation-specific directory

Signed-off-by: Jose R. Gonzalez <josegonzalez89@gmail.com>

skattoju commented on certification/internal/policy/types.go:
```diff
@@ -1,5 +1,9 @@
+
 type Definition struct {
 	ValidatorFunc func(string) (bool, error)
 	Metadata      Metadata
```
any reason we don't include the HelpText field ?

rocrisp reviewed (dismissed):
Looks good to me. 
Thank you for the hard-work on this.

bcrochet reviewed (commented):
/lgtm

samira-barouti commented on certification/certification.go:
```diff
@@ -18,10 +21,13 @@ type Policy interface {
-	"is_ubi_based":        PolicyIsUBIBased,
-	"nonroot":             PolicyRunsAsNonRootUser,
-	"has_required_labels": PolicyImageHasRequiredLabels,
+	"is_ubi_based":                 PolicyIsUBIBased,
```
This needs to be changed based on the upcoming PR.

komish commented:
Rebased and updated to use the new struct-based layout, and placed it in the new implementation specific directory. Net new changes:

- switched all of the structs methods to pointer receivers
- instantiated our policies once and then re-used those to build our mapping.

jomkz reviewed (approved):
/lgtm
